Automattic/wp-calypso
# White glove upsell offer A/B test

## NOTE: Since this PR has become big, I've broken it down to smaller chunks. Check 480-gh-martech for details on the smaller chunks to be reviewed.

## Please do not review this PR, but instead review the smaller chunks

#### Changes proposed in this Pull Request

### This PR implements an A/B test that pitches a "White glove" offer to users who select a free plan and free domain upon signup. More about this is in the experiments post at pbxNRc-gv-p2 and MT at pau2Xa-13X-p2.

1. The offer page looks like

<img width="773" alt="Screenshot 2020-05-28 at 8 01 50 PM" src="https://user-images.githubusercontent.com/1269602/83154714-17e20280-a11e-11ea-9afe-cde6611b16a3.png">

2. Selecting "Buy now" on the offer will add a 1Y Business plan to cart. The product name is modified as "WordPress.com Business (White glove edition)"

**New Checkout**
<img width="557" alt="Screenshot 2020-05-28 at 8 03 12 PM" src="https://user-images.githubusercontent.com/1269602/83154973-5e376180-a11e-11ea-905f-f9616b3d095f.png">

**Old Checkout**
<img width="736" alt="Screenshot 2020-05-28 at 8 03 33 PM" src="https://user-images.githubusercontent.com/1269602/83155016-6d1e1400-a11e-11ea-9b22-551ade0f4e09.png">

3. If purchase is completed, then the receipt email sent to your email inbox will mention the same product name(white glove edition).
<img width="621" alt="Screenshot 2020-05-28 at 8 07 14 PM" src="https://user-images.githubusercontent.com/1269602/83155499-ed447980-a11e-11ea-8358-5fa6ddf206b6.png">

Also, quick start email will say "one-on-one session":

<img width="1000" alt="Screenshot 2020-05-28 at 8 41 17 PM" src="https://user-images.githubusercontent.com/1269602/83159763-f97f0580-a123-11ea-8257-0939186c6635.png">

4. Selecting "No" on the offer will take the user to Customer Home page. There will be a sidebar nudge visible for the first 24 hours of site creation. Clicking the nudge will open the white glove offer page.

<img width="554" alt="Screenshot 2020-05-28 at 8 05 39 PM" src="https://user-images.githubusercontent.com/1269602/83155202-a0f93980-a11e-11ea-865c-a5d6e02c8bd6.png">

5. If you upgrade to "White glove" via the sidebar nudge, then the Thank You page will also mention the product name as "white glove", and will rename "Quick Start" to "one-on-one"

<img width="1500" alt="Screenshot 2020-05-28 at 8 11 49 PM" src="https://user-images.githubusercontent.com/1269602/83156597-3a751b00-a120-11ea-8ef0-c4f67737fbc2.png">

6. The /plans page will rename "Quick Start" to "one-on-one"

<img width="2250" alt="Screenshot 2020-05-28 at 8 18 13 PM" src="https://user-images.githubusercontent.com/1269602/83157346-1cf48100-a121-11ea-8383-fed4fd60a49c.png">

7. The concierge booking form will rename "Quick Start" to "one-on-one"

<img width="1300" alt="Screenshot 2020-05-28 at 8 45 35 PM" src="https://user-images.githubusercontent.com/1269602/83160134-70b49980-a124-11ea-931c-08478783ab0c.png">

#### Testing instructions



Apply D42624-code. For the offer page to show, proxy through a US IP address.

**Scenario 1**(_test offer page is shown and verify offer page UI_)
* Go through the signup flow in the "onboarding" signup flow, select a free plan and free domain
* If you are in `variantShowOffer` of the `whiteGloveUpsell` A/B test, you should see the white glove offer page. Verify that the UI checks out with the [Figma](https://www.figma.com/file/G52TF9nObKGWqwPfDx3Wym/White-Gloves-Interstitial?node-id=50%3A307).
* Verifythat the  UI checks out in mobile screen sizes. 

**Scenario 2** (_verify that the white glove purchase is successful, and the product name in cart says white glove_)

* Assign yourself to the variant of the `whiteGloveUpsell` test. Go through signup flow in the onboarding flow, select free plan and free domain to see the offer page.
* Click "Buy Now" CTA on the offer page and verify that the checkout page says the product name as "WordPress.com Business (White glove edition)"
* Verify that you can complete purchase
* Repeat the above steps for both the old checkout and new checkout. To see th new checkout, you need to proxy through a US IP address and be assigned to the `composite` variation of `showCompositeCheckout` A/B test.

**Scenario 3** (_verify sidebar nudge shows_)

* Assign yourself to the variant of the `whiteGloveUpsell` test. Go through signup flow in the onboarding flow, select free plan and free domain to see the offer page.
* Click No on the offer page. Verify that in the sidebar of Calypso, you see a "White glove" sidebar nudge(screenshot in PR description). 
* Verify that this sidebar nudge is visible only for the first 24 hours after site creation

**Scenario 4** (_More sidebar nudge related tests_)

* Extending Scenario 3 above. Click the sidebar nudge and verify you can see the offer page.
* Click Buy now on the offer page and verify that the product name in cart is "WordPress.com Business (White glove edition)". 
* Complete purchase and verify that you see a Thank You page with white glove mentioned in it, as shown in screenshot in PR description. 
* Sidebar nudge should not be visible if white glove purchase is complete.
* Clicking No on the offer page should take you back to Home page, and the sidebar nudge should still be visible.

**Scenario 5** (verify blog option is set on purchase, removed on cancellation)

* After purchasing the white glove product, verify via wpsh that a blog option called "is_white_glove" is set to true for the site. `return get_blog_option( YOUR_BLOG_ID, 'is_white_glove' );` in wpsh should return `1` 
* Cancel the white glove i.e Business plan purchase, and verify via wpsh that the blog option is removed.

**Scenario 6** (Verify emails mention white glove)

* Purchase a white glove product and verify that the receipt email says "WordPress.com Business (White glove edition)". Check screenshot in PR description.
* Verify that the concierge invitation email says "one-on-one" instead of "Quick Start"

**Scenario 7** (Concierge session booking page should say one-on-one session)
* Purchase a white glove product and head to /me/concierge to book a session.
* Verify that the booking page says "WordPress.com one-on-one Session Scheduler" (check screenshot in description)

**Scenario 8** (offer shown only for EN-US)

* Verify that if you go through the signup flow and select a free plan and free domain, you will not be assigned to the test either for non-EN or EN + non-US.

**Scenario 9** (Payment options)

* When the White Glove product is in your cart, pay for it using PayPal. 
* After purchase completes, verify in wpsh that blog option for white glove is set. That is, `return get_blog_option( YOUR_BLOG_ID, 'is_white_glove' );` in wpsh should return `1`
